src/charm.py

```diff
@@ -228,7 +228,7 @@ def _on_install(self, event: EventBase) -> None:
         Args:
             event: Juju event
         """
-        if not self._is_cpu_compatible():
+        if self._check_cpu_compatibility():
```

Comment:
I liked the previous name better. IMO it reads better.

Reply:
I'm just wondering - isn't it a good case to actually use `defer`? I mean if the CPU is not compatible, there's no way to recover from that. If we defer this even, it will always run before any other events and block the charm. Unless I'm missing something, this seems like a simpler solution comparing to what we have now.
WDYT?